### PR TITLE
feat(qwik): generate pre-optimizer Qwik code

### PR DESCRIPTION
**Pull request by mhevery**
## Description

Add a short description of:
- what changes you made, 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

### CODE REVIEW

**Review by samijaber — COMMENTED** [review]
Can we add an entry for `qwik` to this record: https://github.com/BuilderIO/mitosis/blob/d5ea1d63bdee8862788c150e6724b531a86af02d/packages/core/src/__tests__/shared.ts#L134

Start by adding all of the tests:

```ts
  qwik: [
    BASIC_TESTS,
    ROOT_SHOW_TESTS,
    FORWARD_REF_TESTS,
    MULTI_ON_UPDATE_TESTS,
    FORM_BLOCK_TESTS,
    FOR_SHOW_TESTS,
  ],
```

And comment out all the ones that don't immediately work or the ones that emit clearly problematic code.

The goal is just to have some sort of initial snapshot testing for the final output of `componentToQwik`.

**Comment by mhevery** [comment]
> Can we add an entry for `qwik` to this record:

Yes working on this now.

File: packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap [hunk 1]
```diff
@@ -225,10 +225,597 @@ export const MyComponent = componentQrl(
...
+  useContext,
+  useContextProvider,
+  useStore,
+} from \\"@builder.io/qwik\\";
+import { Injector, MyService, createInjector } from \\"@dummy/injection-js\\";
+const MyBasicComponent = component$((props) => {
+  const state = useStore({
+    name: \\"PatrickJS\\",
+    onChange:
+      \\"@builder.io/mitosis/function:function onChange() {\\\\n  const change = myService.method('change');\\\\n  console.log(change);\\\\n}\\",
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap:294]: This is broken. These functions are not being processed correctly

File: packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap [hunk 2]
```diff
@@ -225,10 +225,597 @@ export const MyComponent = componentQrl(
...
+  useStore,
+} from \\"@builder.io/qwik\\";
+const MyBasicRefComponent = component$((props) => {
+  useScopedStyles$(STYLES);
+  const state = useStore({
+    name: \\"PatrickJS\\",
+    onBlur:
+      \\"@builder.io/mitosis/function:function onBlur() {\\\\n  // Maintain focus\\\\n  inputRef.focus();\\\\n}\\",
+    lowerCaseName:
+      \\"@builder.io/mitosis/function:function lowerCaseName() {\\\\n  return state.name.toLowerCase();\\\\n}\\",
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap:459]: more examples of unprocessed functions in `useStore`

**Review by samijaber — APPROVED** [review]
LGTM. Caught a few missing things, but they can be addressed in followup PRs 👍🏽

File: packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap (hunk 1 above)
Comment (mhevery) [reply packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap]: fixed

File: packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap (hunk 2 above)
Comment (mhevery) [reply packages/core/src/__tests__/__snapshots__/qwik.test.ts.snap]: fixed

**Pull request merged**